backend/cellguide/pipeline/ontology_tree/tree_builder.py

```diff
@@ -347,96 +372,117 @@ def get_ontology_tree_state_per_tissue(self) -> Dict[str, OntologyTreeState]:
-            )
+        results = {
+            tissue: delayed(self._process_tissue__parallel)(tissue)
+            for tissue in self.uberon_by_celltype
+            if tissue.startswith("UBERON:") and " (" not in tissue
+        }
 
-            tissueId = tissue
-            tissue_term = self.uberon_ontology[tissueId]
-            tissue_label = tissue_term.name
-
-            end_nodes = self.uberon_by_celltype[tissue]
-            uberon_ancestors = [i.id for i in tissue_term.superclasses()]
-
-            # filter out hemaotoietic cell types from non-whitelisted tissues
-            uberon_ancestors_in_whitelist = list(set(TISSUES_IMMUNE_CELL_WHITELIST).intersection(uberon_ancestors))
-            if len(uberon_ancestors_in_whitelist) == 0:
-                end_nodes_that_are_not_hematopoietic = [
-                    e
-                    for e in end_nodes
-                    if HEMATOPOIETIC_CELL_TYPE_ID not in [i.id for i in self.ontology[e].superclasses()]
-                ]
-                if len(end_nodes_that_are_not_hematopoietic) == 0:
-                    logger.info(f"Not filtering out immune cell for {tissue_label}")
-                else:
-                    end_nodes = end_nodes_that_are_not_hematopoietic
-            else:
-                logger.info(f"Not filtering out immune cell for {tissue_label}")
+        logger.info(
+            f"Getting ontology tree states for {len(results)} tissues using {CELLGUIDE_PIPELINE_NUM_CPUS} CPUs..."
+        )
+        with ProgressBar():
+            computed_results = compute(*list(results.values()), num_workers=CELLGUIDE_PIPELINE_NUM_CPUS)
+            all_states_per_tissue = {tissue_id: result for tissue_id, result in zip(results.keys(), computed_results)}
 
-            tissue_ct_df = self.tissue_celltypes_df[self.tissue_celltypes_df["tissue_ontology_term_id"] == tissue]
+        return all_states_per_tissue
 
-            # attach cells not in the tissue cell types dataframe
-            # these cells will be added with 0 counts
-            df = tissue_ct_df[["cell_type_ontology_term_id", "n_cells"]]
-            to_attach = pd.DataFrame()
-            to_attach["cell_type_ontology_term_id"] = [
-                i for i in self.all_cell_type_ids if i not in df["cell_type_ontology_term_id"].values
-            ]
-            to_attach["n_cells"] = 0
-            df = pd.concat([df, to_attach], axis=0)
-            # rollup the cell counts
-            df["n_cells_rollup"] = df["n_cells"]
-            df_rollup = rollup_across_cell_type_descendants(df, ignore_cols=["n_cells"])
-            df_rollup = df_rollup[df_rollup["n_cells_rollup"] > 0]
-
-            celltype_counts_in_tissue = dict(
-                zip(
-                    df_rollup["cell_type_ontology_term_id"],
-                    df_rollup[["n_cells", "n_cells_rollup"]].to_dict(orient="records"),
-                )
-            )
+    def _process_tissue__parallel(self, tissueId: str) -> OntologyTreeState:
```

Comment:
this function contains the stuff in the outer loop for processing tissues and is parallelized